src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs

```diff
@@ -288,6 +295,16 @@ public ShapeResult Cache(string cacheId, Action<CacheContext> cache = null)
             return this;
         }
 
+        /// <summary>
+        /// Sets a condition that must return true for the shape to render.
+        /// The condition is only evaluated if the shape has been placed.
+        /// </summary>
+        public ShapeResult IfAsync(Func<Task<bool>> ifAsync)
```

Comment:
Naming is hard ;)

Reply:
About naming, maybe

    public ShapeResult DisplayWhen(Func<Task<bool>> displayPredicateAsync)
    {
        _displayPredicateAsync = displayPredicateAsync;

    public ShapeResult RenderWhen(Func<Task<bool>> renderPredicateAsync)
    {
        _renderPredicateAsync = renderPredicateAsync;

Reply:
I like `RenderWhen` thanks @jtkech